### PR TITLE
Remove service datasets

### PR DESCRIPTION
**Pull request by jalvz**
## Motivation/summary

We need to remove service names from index names, as they need to manage exactly what Fleet expects for data streams to be created.

## Checklist

- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).
- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)

I have considered changes for:
- [ ] documentation
- [ ] logging (add log lines, choose appropriate log selector, etc.)
- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))
- [ ] automated tests (add tests for the code changes, all [**unit** tests](https://github.com/elastic/apm-server/blob/master/TESTING.md) pass locally)
- [ ] telemetry
- [ ] Elasticsearch Service (https://cloud.elastic.co)
- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)
- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)

### CODE REVIEW

**Comment by axw** [comment]
The main changes looks fine, but what's up with the pipeline removal? If it's not related to this change, can you please separate that into another PR?

**Comment by jalvz** [comment]
The pipeline removal is a diff artifact, and looks "re-added" here: apmpackage/apm/0.1.0/data_stream/profile_metrics/elasticsearch/ingest_pipeline/default.json. But it is the same with the name fixed.

The reason for the pipeline changes is that the profiles one had the wrong name (`s/profiles-apm-0.1.0/metrics-apm.profiling-0.1.0`) which I blamed on poor data stream naming, so I took the chance to rename the data streams as well.

I made package and code changes into separated commits, but happy to move one out to another PR if you prefer.

**Review by axw — APPROVED** [review]
> The pipeline removal is a diff artifact, and looks "re-added" here: apmpackage/apm/0.1.0/data_stream/profile_metrics/elasticsearch/ingest_pipeline/default.json. But it is the same with the name fixed.
>
> The reason for the pipeline changes is that the profiles one had the wrong name (s/profiles-apm-0.1.0/metrics-apm.profiling-0.1.0) which I blamed on poor data stream naming, so I took the chance to rename the data streams as well.
>
> I made package and code changes into separated commits, but happy to move one out to another PR if you prefer.

I see. Since you're reusing the constant now, it makes sense to have them in the same PR. Thanks for the explanation.

File: datastreams/servicename.go [hunk 1]
```diff
@@ -24,6 +24,8 @@ import "strings"
 //
 // Concretely, this function will lowercase the string and replace any
 // reserved characters with "_".
+//
+// TODO: use when Fleet supports variables in data streams
```
[the remark this answers is not in the document]
Comment (jalvz) [reply datastreams/servicename.go:28]:
```suggestion
// TODO: use when Fleet supports variables in data streams (see #4492)
```

**Comment by simitt** [comment]
Not sure this needs any manual testing/how to test this manually. I'll leave it there fore now @jalvz can you please leave a comment on what to test here if anything should be done, otherwise add the `test_plan_skip` label - thank you.

**Comment by jalvz** [comment]
Probably the test of this is covered already by some other issue. Basically, data streams should be created when data is ingested. It was not the case before this.

**Pull request merged**